pkgs/by-name/ol/ollama/package.nix

```
@@ -199,13 +201,20 @@ goBuild {
       make ${dist_cmd} -j $NIX_BUILD_CORES
     '';
 
+  postInstall = lib.optionalString stdenv.hostPlatform.isLinux ''
+    # copy libggml_*.so and runners into lib
+    # https://github.com/ollama/ollama/blob/v0.4.4/llama/make/gpu.make#L90
+    mkdir -p $out/lib
+    cp -r dist/*/lib/* $out/lib/
```

as per https://github.com/NixOS/nixpkgs/pull/373234#issuecomment-2592206820 it seems there are no files matching `dist/*/lib/*` on aarch64-linux


```suggestion
    # on some architectures there are no files matching dist/*/lib/*
    cp -r dist/*/lib/* $out/lib/ || :
```

---

did you build for cuda or rocm?

---

i did not build anything, the comment https://github.com/NixOS/nixpkgs/pull/373234#issuecomment-2592206820 is from @GaetanLepage

---

you misread, ollama-cuda is fine, it's ollama which is broken on aarch64-linux, because there are no files in `dist/*/lib/*` - can you just apply my suggestion please?

and also change the commit message to read `ollama: 0.5.4 -> 0.5.5` (`->` instead of `>`)

---

I don't have the power to make the changes here, but if you see https://github.com/abysssol/ollama-flake/pull/17 Following abysssol's suggestion to  use `(enableRocm || enableCuda)` instead of `stdenv.hostPlatform.isLinux`  may resolve this.

---

That change is suboptimal because then on x86_64 CPU ollama will be missing the cpu_avx and cpu_avx2 runners.